sklearn/ensemble/_gb.py

```diff
@@ -362,6 +368,10 @@ def _check_initialized(self):
     def _warn_mae_for_criterion(self):
         pass
 
+    @abstractmethod
+    def _warn_mse_for_criterion(self):
+        pass
```

Comment:
In the `mse` case the warning message is exactly the same, so we can include the warning inline with the `self.criterion == 'mse'` check.